caikit/runtime/client/utils.py

```diff
@@ -79,4 +120,16 @@ def construct_requests_session(
     if timeout:
         session.params["timeout"] = timeout
 
+    # Mount retry object if options were provided
+    if retries:
+        default_status_codes = list(Retry.RETRY_AFTER_STATUS_CODES) + [500, 502, 504]
```

Comment:
Probably not for this PR, but I do see that the builtin `http` package holds a `http.HTTPStatus` enum that we should probably start using everywhere.

https://docs.python.org/3/library/http.html#http.HTTPStatus

Reply:
Oh I didn't realize that was in the standard library! It would be neat to refactor the code to use them